qiskit/visualization/pulse_v2/style/stylesheet.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+r"""
+Stylesheet for pulse drawer.
+
+The general stylesheet template `QiskitPulseStyle` is initialized with the hard-corded
+default values in the `default.json` file. This instance is generated when the pulse drawer
+module is loaded so that every lower modules can access to the information.
+
+The `QiskitPulseStyle` takes nested python dictionary and stores the settings under
+flattened dictionary keys such as `formatter.color.background`.
```

Comment:
Why do we store it in a flattened form?

Reply:
Because we may want to avoid nested dict access like `style['formatter']['color']['background']`. I tried to implement attribute-like access to python dictionary but cannot find the solution to cope with IDE autocomplete.

Reply:
@1ucian0 are you ok with this configuration style? I worry that we should consider implementing a standardized style across visualization modules for Qiskit.

Reply:
I updated to simplify the stylesheet framework as you suggested. Standardized style sounds good. Perhaps I need to consider some requirements of circuit drawer. I think the advantage of dict based stylesheet is the stable user interface. We can easily raise deprecation warning without keeping duplicated arguments in the interface function.

https://github.com/Qiskit/qiskit-terra/blob/5ea983a564ac576186a8cd6faf4b8131ff160ff0/qiskit/visualization/pulse_v2/style/stylesheet.py#L137-L156

Since we don't have interface itself in this PR it may be difficult to imagine how it works. I assume something like:

```python
drawer_interface(program, style=library.IqxStandard())
```

The stylesheet instance `QiskitPulseStyle` is internally initialized as `PULSE_STYLE` at the top module level, i.e. `__init__`, with the default style dict (so that it can be accessed by all lower modules). Then it is overwritten by the input style:

```python
# in some interface module
import PULSE_STYLE

def drawer_interface(program, style):
  PULSE_STYLE.update(style)
```

If the user specify the custom style dictionary and it includes deprecated key, `QiskitPulseStyle` issues a deprecation warning and automatically replace the key by checking the lookup table.

Reply:
> @1ucian0 are you ok with this configuration style? I worry that we should consider implementing a standardized style across visualization modules for Qiskit.

I'm not much into the MPL circuit drawer. Probably a question for @mtreinish or @nonhermitian ...

Reply:
Yeah, it would be nice to have a Qiskit-wide stylesheet that can be loaded a la seaborn, e.g. https://seaborn.pydata.org/generated/seaborn.set_style.html#seaborn.set_style

It has also been commented several times that the Qiskit visualizations are not "publication quality", so a global stylesheet might help to address that as well.

Reply:
Thanks. I'll talk with @mtreinish later about the actual implementation of the qiskit-wide style sheet. I think seaborn has the correct inheritance of matplotlib but we may have stylesheets which is more specialized to the circuit or pulse visualization.

As you mentioned I needed to modify the visualization code to sophisticate the output image for publication. This series of PRs are aiming at solving the issues and giving the users to much more flexibility to customize the output. Since the stylesheet is very flexible part, so I want to merge this PR first and update the stylesheet framework in the later PRs.